pkg/hooks/v1alpha2/api.proto

```diff
@@ -20,13 +20,19 @@ message OnDefineDomainResult {
 }
 
 message PreCloudInitIsoParams {
-  // cloudInitData is an object of CloudInitNoCloudSource encoded as JSON
-  bytes cloudInitData = 1;
+  // cloudInitNoCloudSource is an object of CloudInitNoCloudSource encoded as JSON
+  // This is a legacy field to ensure backwards compatibility. New code should use cloudInitData instead.
+  bytes cloudInitNoCloudSource = 1;
```

Comment:
can the name be changed here? I thought that would break compatiblity.

Reply:
Tbh, I am no expert on protobuf3. But as I understand https://developers.google.com/protocol-buffers/docs/proto3#updating it should be ok to rename fields as long as the number stays the same. I will continue to test this to ensure that the code is backwards compatible.

Reply:
Yeah in protobuf it's ok to change names.

Reply:
Yes, it is ensured that the code is backwards compatible.

As you can see in the `pkg/hooks/manager.go` PreCloudInitIso method, a CloudInitNoCloudSource object is still created and passed through PreCloudInitIsoParams into the sidecar hook under the new name (but with the old id). So old sidecar hooks will receive data on that param just the way they expect, whereas new sidecar hooks can use the cloudInitData param with the new id.

In case the returned cloud-init data from the sidecar is not a valid, the function checks if a CloudInitNoCloudSource object was returned and if yes transforms it into a valid cloud-init data object.

This way backwards compatibility is ensured when passing params into the sidecar hook and when receiving a result from a sidecar look.